File: Data Structures & Algorithms/longest-consecutive-sequence/submission-1.cpp
class Solution {
public:
    int longestConsecutive(vector<int>& nums) {
        
        int start=-1;
        int n=nums.size();
        if(n==0){
            return 0;
        }
        unordered_map<int,int>mp; 
        for(int i=0;i<nums.size();i++){
           mp[nums[i]]=i;

        }
        int cnt=1;
        int vis[n]={0};
        int maxi=INT_MIN;
        for(int i=0;i<nums.size();i++){
            int num=nums[i];
            if(!vis[i]){
                vis[i]=1;
                while(mp.find(num+1)!=mp.end()){
                 vis[mp[num+1]]=1;
                 cnt++;
                 num=num+1;

                }
             maxi=max(maxi,cnt);
            }
            cnt=1;

            
        
        } 
        return maxi;
    }
};

File: Data Structures & Algorithms/largest-3-same-digit-number-in-string/submission-1.java
class Solution {
    public String largestGoodInteger(String num) {
        int []number=new int[10];
        int maxi=-1;
        for(int i=0;i<num.length()-2;i++){
            
            String n=num.substring(i,i+3);
            for(int j=0;j<3;j++){
               number[n.charAt(j)-'0']++;
            }
            int cnt=0;
            for(int j=0;j<10;j++){
                if(number[j]!=0) cnt++;
                if(cnt>1){
                    break;
                }
            }
            if(cnt==1){
                maxi=Math.max(maxi,Integer.parseInt(n));
            }

            Arrays.fill(number,0);



        }
        if(maxi==-1) return "";
        if(maxi==0) return "000";
        return String.valueOf(maxi);

           
    }
}